File: lesson_19/src/main/java/org/lesson19/MySessionContextClass.java
package org.lesson19;

import org.hibernate.context.internal.ThreadLocalSessionContext;
import org.hibernate.engine.spi.SessionFactoryImplementor;

public class MySessionContextClass extends ThreadLocalSessionContext {
    public MySessionContextClass(SessionFactoryImplementor factory) {
        super(factory);
    }

    @Override
    protected boolean isAutoCloseEnabled() {
        return false;
    }

    @Override
    protected boolean isAutoFlushEnabled() {
        return true;
    }
}

File: lesson_19/src/main/kotlin/org/lesson19kt/MySessionContextClass.kt
package org.lesson19kt

import org.hibernate.context.internal.ThreadLocalSessionContext
import org.hibernate.engine.spi.SessionFactoryImplementor

class MySessionContextClass(factory: SessionFactoryImplementor) : ThreadLocalSessionContext(factory) {
    override fun isAutoCloseEnabled() = false
    override fun isAutoFlushEnabled() = true
}